.github/workflows/ci.yml

```diff
@@ -320,6 +320,64 @@ jobs:
         # reason for excluding https://github.com/napi-rs/napi-rs/issues/2200
         run: cargo test --workspace --exclude rspack_binding_options --exclude rspack_node -- --nocapture
 
+  rust_test_codecov:
+    name: Rust Test Coverage
+    needs: [rust_changes]
+    if: ${{ needs.rust_changes.outputs.changed == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Pnpm Cache # Required by some tests
+        uses: ./.github/actions/pnpm-cache
+
+      - name: Install Rust Toolchain
+        uses: ./.github/actions/rustup
+        with:
+          save-cache: ${{ github.ref_name == 'main' }}
+          shared-key: check
+
+      - name: Install llvm-tools-preview
+        run: rustup component add llvm-tools-preview
+      - uses: cargo-bins/cargo-binstall@v1.10.2
+      - run: cargo binstall --no-confirm cargo-llvm-cov
+
+      - name: Run test coverage
+        # reason for excluding https://github.com/napi-rs/napi-rs/issues/2200
+        run: cargo llvm-cov --workspace --exclude rspack_binding_options --exclude rspack_node --lcov --output-path lcov.info
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: codecov
+          path: lcov.info
+
+  rust_test_codecov_upload:
```

Comment:
Why split rust_test_codecov and rust_test_codecov_upload, can we merge them?

Reply:
`# codecov often fails, use another workflow for retry` 
https://github.com/web-infra-dev/rspack-resolver/blob/16eb56eef68d651edec703b94022ff7f3e46d585/.github/workflows/codecov.yml#L44

Migrated from `rspack-resolver` The generated file is huge, and the api of codecov does not seem to be very stable.